pkg/cloud/services/compute/instance.go

```diff
@@ -382,38 +382,64 @@ func (s *Service) getSecurityGroups(securityGroupParams []infrav1.SecurityGroupP
 
 func (s *Service) getServerNetworks(networkParams []infrav1.NetworkParam) ([]infrav1.Network, error) {
 	var nets []infrav1.Network
+
+	addSubnet := func(netID, subnetID string) {
+		nets = append(nets, infrav1.Network{
+			ID: netID,
+			Subnet: &infrav1.Subnet{
+				ID: subnetID,
+			},
+		})
+	}
+
+	addSubnets := func(networkParam infrav1.NetworkParam, netID string) error {
+		if len(networkParam.Subnets) == 0 && netID != "" {
+			addSubnet(netID, "")
+			return nil
+		}
+
+		for _, subnet := range networkParam.Subnets {
+			// Don't lookup subnet if it was specified explicitly by UUID
+			if subnet.UUID != "" {
+				addSubnet(netID, subnet.UUID)
+				continue
```

Comment:
nit pick, IMO continue is just a more confusing way to write if {} else {}, but its a very minor nit pick

Reply:
Yeah. Sometimes it's worth it if the following block is longer, but I agree an else block would have been clearer here.

I'll change this if I have to update the patch for any other reason.

Reply:
I updated this.